src/iov.c

```diff
@@ -79,8 +79,13 @@ void ofi_consume_iov_desc(struct iovec *iov, void **desc,
 			sizeof(*desc) * (*iov_count - i));
 	*iov_count -= i;
 out:
-	iov[0].iov_base = (uint8_t *)iov[0].iov_base + to_consume;
-	iov[0].iov_len -= to_consume;
+	if (to_consume < iov[0].iov_len) {
+		iov[0].iov_base = (uint8_t *)iov[0].iov_base + to_consume;
+		iov[0].iov_len -= to_consume;
+	} else if (*iov_count) {
```

Comment:
This check implies that iov_count can be 0. But we've already assumed that it is 1 by dereferencing iov[0] above.

I don't think we need to handle iov_count = 0.  Just assert that it's >= 1.

We also don't need to set iov[0].iov_len =0.  If iov_count = 0 on return, then the user shouldn't access iov[0].

Reply:
Done.  I also explicitly added the asserts.